build.cake

```diff
@@ -0,0 +1,497 @@
+	if(BuildSystem.IsRunningOnAppVeyor)
+	{
+		Information("Upload artifact to AppVeyor...");
+		BuildSystem.AppVeyor.UploadArtifact(_zipFilePath);
+		var msiFile = @".\GitTfs.Setup\GitTfs.Setup.msi";
+		if(FileExists(msiFile))
+			BuildSystem.AppVeyor.UploadArtifact(msiFile);
+	}
+});
+
+void DisplayAuthTokenErrorMessage()
+{
+	var errorMessage = @"Please create a file 'PersonalTokens.config' containing your authentication tokens
+See the file 'PersonalTokens.config.example' for the format and content.";
+
+	throw new Exception(errorMessage);
+}
+
+string ReadToken(string tokenKey, string tokenRegexFormat = null)
+{
+	var authTargetsFile = "PersonalTokens.config";
+
+	Information("Reading token..." + tokenKey);
+
+	if(!FileExists(authTargetsFile))
+		DisplayAuthTokenErrorMessage();
+
+	var personalToken = System.IO.File.ReadAllLines(authTargetsFile).FirstOrDefault(l => l.StartsWith(tokenKey + "="));
+	if(personalToken == null)
+		DisplayAuthTokenErrorMessage();
+
+	personalToken = personalToken.Trim();
+	personalToken = personalToken.Substring(tokenKey.Length+1,personalToken.Length-tokenKey.Length-1);
+	if(tokenRegexFormat == null)
+		return personalToken;
+	
+	var regexToken = new System.Text.RegularExpressions.Regex(tokenRegexFormat);
+	if(!regexToken.IsMatch(personalToken))
+		DisplayAuthTokenErrorMessage();
+	return personalToken;
+}
+
+string GetChocolateyToken()
+{
+	var chocolateyToken = Argument("chocolateyToken", "");
+	if(!string.IsNullOrEmpty(chocolateyToken))
+		return chocolateyToken;
+	return ReadToken("Chocolatey");
+}
+
+string GetGithubUserAccount()
+{
+	var gitHubOAuthToken = Argument("gitHubUserAccount", "");
+	if(!string.IsNullOrEmpty(gitHubOAuthToken))
+		return gitHubOAuthToken;
+	return ReadToken("GitHubUserAccount");
+}
+
+string GetGithubAuthToken()
+{
+	var gitHubOAuthToken = Argument("gitHubToken", "");
+	if(!string.IsNullOrEmpty(gitHubOAuthToken))
+		return gitHubOAuthToken;
+	return ReadToken("GitHub", "^[0-9a-f]{40}$");
+}
+
+string ReadReleaseNotes()
+{
+	if(!FileExists(ReleaseNotesPath))
+		return string.Empty;
+	return System.IO.File.ReadAllText(ReleaseNotesPath);
+}
+
+Octokit.GitHubClient GetGithubClient()
+{
+	var githubToken = GetGithubAuthToken();
+	var client = new Octokit.GitHubClient(new Octokit.ProductHeaderValue("git-tfs-releasing"));
+	var tokenAuth = new Octokit.Credentials(githubToken);
+	client.Credentials = tokenAuth;
+	return client;
+}
+
+Task("TriggerRelease").Description("Trigger a release from the AppVeyor build server")
+	.Does(() =>
+{
+	var httpClient = new System.Net.Http.HttpClient();
+	//AppVeyor build data to trigger the git-tfs build + parameters passed to the release build 
+	var content = @"{
+accountName: 'pmiossec',
+projectSlug: 'git-tfs-v2qcm',
+branch: 'master',
+environmentVariables: {
+ target: 'AppVeyorRelease',
+ chocolateyToken: '"+ GetChocolateyToken() + @"',
+ gitHubUserAccount: '"+ GetGithubUserAccount() + @"',
+ gitHubToken: '" + GetGithubAuthToken() + @"'
```

Comment:
Personally, I am not in favor of posting my token to appveyor, because I don't know whether it's going to be handled as sensitive data or not. Do they have a policy available about it? Also, if this is going to be the official release process, can we move the appveyor build so that it's also called "git-tfs/git-tfs" there?

As a first step, before making appveyor do the build, could the release build happen on a local workstation for now? once the build process is pretty solid, then we can look into putting it on appveyor, or something else.